minimint/src/lib.rs

```diff
@@ -218,12 +218,12 @@ impl MinimintServer {
     }
 }
 
-struct CloneRngGen<T: RngCore + CryptoRng + Clone + Send>(Mutex<T>);
+struct CloneRngGen<T: RngCore + CryptoRng + Clone + Send + Sync>(T);
 
-impl<T: RngCore + CryptoRng + Clone + Send> RngGenerator for CloneRngGen<T> {
+impl<T: RngCore + CryptoRng + Clone + Send + Sync> RngGenerator for CloneRngGen<T> {
     type Rng = T;
 
     fn get_rng(&self) -> Self::Rng {
-        self.0.lock().unwrap().clone()
+        self.0.clone()
```

Comment:
I had a look at `OsRngInner` and it seems reasonable to just generate a new one every time. Maybe instead of `CloneRngGen<T>` we could just have a `OsRngGen`.